### PR TITLE
US013 -  Adding a Capability

### PR DESCRIPTION
**Pull request by anthony-raybould-kainos**
Implements the UI for creating a capability at the URL capabilities/create.

image

image

### CODE REVIEW

**Review by ManpreetM01 — APPROVED** [review]

File: src/service/capabilityService.ts [hunk 1]
```diff
@@ -15,5 +16,29 @@ export const capabilityService = {
...
+    async createCapability(createCapabilityRequest : CreateCapabilityRequest, token? : string): Promise<void> {
+
+        if (!token) throw new Error("You are not logged in (no token provided)")
+
+        try {
+            const response = await axios.post(`${process.env.API_URL}api/capabilities/`, createCapabilityRequest, { 
+                headers: { Authorization: `Bearer ${token}` }
+            });
+
+            if (response.status === 200) {
```
Comment (RobertCrabtreeKainos) [inline src/service/capabilityService.ts:30]: should this be 201 for created rather than OK 200?

File: src/service/capabilityService.ts [hunk 2]
```diff
@@ -15,5 +16,29 @@ export const capabilityService = {
...
+                return;
+            } 
+            if(response.status === 400) {
+                throw new Error(`Bad request. ${response.data.errorMessage}`);
+            }
+            else {
+                throw new Error(response.data.errorMessage);
+            }
+        } catch (e) {
+            console.error(e);
```
Comment (RobertCrabtreeKainos) [inline src/service/capabilityService.ts:37]: user wouldnt b able to see Error that thrown on 37. And error message would be lost and returned to "Failed creating capability" maybe add the error message to the caught and thrown new error on line 42

File: src/service/capabilityService.ts (hunk 1 above)
Comment (RobertCrabtreeKainos) [inline src/service/capabilityService.ts:30]: should this be 201 for created rather than OK 200?

File: views/create-capability-success.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    Create Capability
+{% endblock %}
+
+{% block content %}
+    
+<h5>Capability "{{ capabilityName }}" created successfully</h5>
+
+<a href="/" class=" bg-kainos-blue btn btn-primary text-white">Home</a>
+
+{% endblock %}
```
Comment (RobertCrabtreeKainos) [inline views/create-capability-success.html:13]: if we dont have links at top of page user wont be able to go to any pages after this confirmation page

**Review by TomBillingtonUK — APPROVED** [review]

**Pull request merged**